lib/viewAction.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { Node } from './files/node.ts'
+import { View } from './navigation/view.ts'
+
+export interface ViewActionData {
+	/** Unique ID */
+	id: string
+
+	/** Translated name of the action */
+	displayName: string
+
+	/** Translated title of the action */
+	title?: string
+
+	/** Raw svg string */
+	iconSvgInline: string
+
+	/**
+	 * Returns true if this action shoud be shown
+	 *
+	 * @param nodes The nodes in the current directory
+	 */
+	show?: (view: View, nodes: Node[]) => boolean
+
+	/**
+	 * Returns true if the action should be disabled
+	 *
+	 * @param nodes The nodes in the current directory
+	 */
+	disabled?: (view: View, nodes: Node[]) => boolean
+
+	/**
+	 * Function to execute
+	 *
+	 * @param nodes The nodes in the current directory
+	 */
+	exec: (view: View, nodes: Node[]) => Promise<void>
+}
+
+export class ViewAction {
+
+	private _action: ViewActionData
+
+	constructor(action: ViewActionData) {
+		this.validateAction(action)
+		this._action = action
+	}
+
+	get id() {
+		return this._action.id
+	}
+
+	get displayName() {
+		return this._action.displayName
+	}
+
+	get title() {
+		return this._action.title
+	}
+
+	get iconSvgInline() {
+		return this._action.iconSvgInline
+	}
+
+	get show() {
+		return this._action.show
+	}
+
+	get disabled() {
+		return this._action.disabled
+	}
```

Comment:
I would prefer to stay close to the other actions.

`enabled` should be enough, no?

Reply:
Implemented it this way to allow displaying a disabled button in contrast to FileAction where `enabled` does not reflect the disabled state but whether it should be rendered or not

I think it makes sense to allow rendering a disabled button in this context vs. removing the button completely after e.g. emptying the trashbin and there are no more items to delete

Reply:
Any design thoughts @marcoambrosini?

Reply:
I agree with @skjnldsv here, what would be the real use case?
Having is visible but disabled is a pattern we should remove everywhere from our code, it is not accessible as disabled elements are purely visual. So I think we should stick with `enabled`.

Reply:
Agreed :)